app/uk/gov/hmrc/fileupload/controllers/EnvelopeController.scala

```diff
@@ -50,21 +46,27 @@ class EnvelopeController(withBasicAuth: BasicAuth,
 
   def create() = Action.async(jsonBodyParser[CreateEnvelopeRequest]) { implicit request =>
     def envelopeLocation = (id: EnvelopeId) => LOCATION -> s"${ request.host }${ uk.gov.hmrc.fileupload.controllers.routes.EnvelopeController.show(id) }"
-    val command = CreateEnvelope(nextId(), request.body.callbackUrl, request.body.expiryDate, request.body.metadata)
+    val constraints = Envelope.setConstraints(request.body.constraints)
```

Comment:
Envelope.setConstraints looks like a setter method that mutates an object, it's not happening here but confused me...